### DISCUSSION TITLE
Support for other validators

### DISCUSSION THREAD

**Discussion by CodeWithShreyans** [next-safe-action/next-safe-action#45] [Ideas]
It would be pretty nice to have modular support for using alternate validators like [Valibot](https://valibot.dev/).

**Comment by TheEdoRan** [comment]
It would be cool, but a huge amount of work too, since the lib in its current state is deeply tied to Zod parsing and types. Supporting multiple validators means switching to a modular design and updating examples/documentation for new supported validators. Let's see what others think about this suggestion.

**Reply by TheEdoRan** [reply]
[TypeSchema](https://github.com/decs/typeschema) looks very promising!

**Comment by TheEdoRan** [comment]
next-safe-action v6 supports a wide range of validation libraries, thanks to [TypeSchema](https://typeschema.com/#coverage). More information [here](https://github.com/TheEdoRan/next-safe-action/discussions/47) and on the updated [documentation website](https://next-safe-action.dev).
